lib/ansible/module_utils/utm_utils.py

```diff
@@ -0,0 +1,210 @@
+    just initialize this UTMModule class and define the Payload fields that are needed for your module.
+    See the other modules like utm_aaa_group for example.
+    """
+
+    def __init__(self, argument_spec, bypass_checks=False, no_log=False, check_invalid_arguments=None,
+                 mutually_exclusive=None, required_together=None, required_one_of=None, add_file_common_args=False,
+                 supports_check_mode=False, required_if=None):
+        default_specs = dict(
+            utm_host=dict(type='str', required=True),
+            utm_port=dict(type='int', default=4444),
+            utm_token=dict(type='str', required=True, no_log=True),
+            utm_protocol=dict(type='str', required=False, default="https", choices=["https", "http"]),
+            validate_certs=dict(type='bool', required=False, default=True),
+            state=dict(default='present', choices=['present', 'absent', 'info'])
+        )
+        super(UTMModule, self).__init__(self._merge_specs(default_specs, argument_spec), bypass_checks, no_log,
+                                        check_invalid_arguments, mutually_exclusive, required_together, required_one_of,
+                                        add_file_common_args, supports_check_mode, required_if)
+
+    def _merge_specs(self, default_specs, custom_sepcs):
+        result = default_specs.copy()
+        result.update(custom_sepcs)
+        return result
+
+
+class UTM:
+
+    def __init__(self, module, endpoint, change_relevant_keys):
+        """
+        Initialize UTM Class
+        :param module: The ansible module
+        :param endpoint: The corresponing endpoint to the module
+        :param change_relevant_keys: The keys of the object to check for changes
+        """
+        self.module = module
+        self.request_url = module.params.get('utm_protocol') + "://" + module.params.get('utm_host') + ":" + str(
+            module.params.get('utm_port')) + "/api/objects/" + endpoint + "/"
+
+        """
+        The change_relevant_keys will be checked for changes to determine whether the object needs to be updated
+        """
+        self.change_relevant_keys = change_relevant_keys
+        self.module.params['url_username'] = 'token'
+        self.module.params['url_password'] = module.params.get('utm_token')
+        if all(elem in self.change_relevant_keys for elem in module.params.keys()):
+            raise UTMModuleConfigurationError(
+                "The keys " + str(self.change_relevant_keys) + " to check are not in the modules keys:\n" + str(
+                    module.params.keys()))
+
+    def execute(self):
+        try:
+            if self.module.params.get('state') == 'present':
+                self._add()
+            elif self.module.params.get('state') == 'absent':
+                self._remove()
+            elif self.module.params.get('state') == 'info':
+                self._info()
```

Comment:
@bcoca noted the use of state=info in today's meeting... I think a year ago, or so, we discussed using a separate module named with an _info suffix (like sophos_utm_info.py) for things that are just for gathering info about something which doesn't relate to the host it is run on.  (You also thought you might turn that portion into a lookup plugin.  That would also be fine.  The difference is just that a lookup can only be run o nthe controller whereas a module can be used on either the controller or a remote host).

Reply:
Maybe we should go for the _info variant then. I will think about it